src/server/pfs/s3/s3.go

```diff
@@ -113,29 +113,59 @@ func Router(driver Driver, clientFactory ClientFactory) *mux.Router {
 	return s3Server.Router()
 }
 
-// Server runs an HTTP server with an S3-like API for PFS. This allows you to
+// S3Server wraps an HTTP server with an S3-like API for PFS. This allows you to
 // use s3 clients to access PFS contents.
 
-// Modifications to `routerMap` allow dynamic changes for the mapping between s3 endpoints and API configurations
-func Server(port uint16, defaultRouter *mux.Router, routerMap map[string]*mux.Router, routersLock *sync.Mutex) *http.Server {
+// In addition to providing the http server itself, S3Server exposes methods
+// to configure handlers (mux.Routers) corresponding to different request URI hostnames.
+// This way one http Server can respond differently and accordingly to each specific job.
+type S3Server struct {
+	HttpServer  *http.Server
```

Comment:
making this an embedded struct would make some of the http-related call sites simpler

Reply:
I like it!